src/watertap_contrib/reflo/analysis/multiperiod/ltmed_vagmd_semibatch/MED_VAGMD_flowsheet.py

```diff
@@ -0,0 +1,406 @@
+    if module_type == "AS7C1.5L" and md_high_brine_salinity:
+        cooling_system_type = "closed"
+        feed_flow_rate = 1100  # L/h
+        evap_inlet_temp = 80  # deg C
+        cond_inlet_temp = 25  # deg C
+
+    fs.vagmd = VAGMDSurrogateBase(
+        property_package_seawater=fs.liquid_prop,
+        property_package_water=fs.vapor_prop,
+        module_type=module_type,
+        high_brine_salinity=md_high_brine_salinity,
+        cooling_system_type=cooling_system_type,
+    )
+
+    # Specify evaporator inlet temperature
+    fs.vagmd.evaporator_in_props[0].temperature.fix(evap_inlet_temp + 273.15)
+
+    # Identify cooling system type
+    # Closed circuit, in which TCI is forced to be constant and the cooling water temperature can be adjusted.
+    if cooling_system_type == "closed":
+        fs.vagmd.condenser_in_props[0].temperature.fix(cond_inlet_temp + 273.15)
+    # Open circuit, in which cooling is available at a constant water temperature and condenser inlet temperature varies.
+    else:  # "open"
+        fs.vagmd.cooling_in_props[0].temperature.fix(cooling_inlet_temp + 273.15)
+
+    return
+
+
+def fix_dof_and_initialize(
+    m,
+    outlvl=idaeslog.WARNING,
+):
+    """Fix degrees of freedom and initialize the flowsheet
+    This function fixes the degrees of freedom of each unit and initializes the entire flowsheet.
+    Args:
+        m: Pyomo `Block` or `ConcreteModel` containing the flowsheet
+        outlvl: Logger (default: idaeslog.WARNING)
+    """
+
+    calculate_scaling_factors(m)
+    # Add scaling factors for variable and constraints constructed in the flowsheet
+    if get_scaling_factor(m.fs.volume_in_tank) is None:
+        set_scaling_factor(m.fs.volume_in_tank, 1e2)
+        set_scaling_factor(m.fs.volume_in_tank_previous, 1e2)
+
+    if get_scaling_factor(m.fs.dt) is None:
+        set_scaling_factor(m.fs.dt, 1e-1)
+
+    m.fs.med.initialize()
+    m.fs.S1.initialize()
+    # Specify feed flow state properties
+    m.fs.vagmd.feed_props.calculate_state(
+        var_args={
+            ("flow_vol_phase", "Liq"): pyunits.convert(
+                600 * pyunits.L / pyunits.h,
+                to_units=pyunits.m**3 / pyunits.s,
+            ),
+            ("conc_mass_phase_comp", ("Liq", "TDS")): 70,
+            ("temperature", None): 35 + 273.15,
+            ("pressure", None): 101325,
+        },
+        hold_state=True,
+    )
+    m.fs.vagmd.initialize()
+
+    calculate_variable_from_constraint(m.fs.dt, m.fs.eq_dt)
+
+    m.fs.M1.remained_liquid_state.calculate_state(
+        var_args={
+            ("flow_vol_phase", "Liq"): 50 / 1000 / m.fs.dt.value,  # m3/s
+            ("conc_mass_phase_comp", ("Liq", "TDS")): 70 * pyunits.kg / pyunits.m**3,
+            ("temperature", None): 35.7 + 273.15,
+            ("pressure", None): 101325,
+        },
+        hold_state=True,
+    )
+
+    m.fs.M1.initialize()
+    m.fs.S2.initialize()
+
+    m.fs.volume_in_tank_previous.fix(0)
+
+    return
```

Comment:
Is there any reason for a return here?

Reply:
Nope! It's removed now.

And thanks a lot for the modifications/suggestions in the doc!